mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+"""
+This module provides pre-configured HTTP requests
+to consume a desired endpoint.
+"""
+
+from __future__ import annotations
+import sys
+import os
+import logging
+from copy import deepcopy
+from enum import Enum
+from requests import Request, Response, Session
+
+
+class Roles(Enum):
+    """
+    User roles available in McM
+    """
+
+    USER = "user"
+    GENERATOR_CONTACT = "generator_contact"
+    GENERATOR_CONVENER = "generator_convener"
+    PRODUCTION_MANAGER = "production_manager"
+    ADMINISTRATOR = "administrator"
+
+
+class APIRequest:
+    """
+    Provides some utilities to perform HTTP request for consuming
+    endpoints in the McM application impersonating a user or interacting
+    with the database.
+    """
+
+    def __init__(
+        self,
+        mcm_couchdb_url: str = "",
+        mcm_couchdb_credential: str = "",
+        mcm_couchdb_lucene_url: str = "",
+        mcm_application_url: str = "",
+        mockup: bool = False,
+    ) -> None:
+        """
+        Instantiate the API request client.
+
+        Attributes:
+            mcm_couchdb_url (str): McM database URL.
+                If it is not provided, it will be tried to retrieve it from
+                `MCM_COUCHDB_URL`.
+            mcm_couchdb_credential (str): Credentials to write data in
+                CouchDB. If it is not provided, it will be tried to retrieve it from
+                `COUCH_CRED`
+            mcm_couchdb_lucene_url (str): McM search engine URL.
+                If it is not provided, it will be tried to retrieve it from
+                `MCM_LUCENE_URL`.
+            mcm_application_url (str): McM web application URL.
+                If it is not provided, it will be tried to retrieve it from
+                - Host: MCM_HOST
+                - Port: MCM_PORT
+            mockup (bool): If True, the creation/validation of mock users
+                will not be perform
+        Raises:
+            RuntimeError: In case some of the required attributes finish to be empty.
+        """
+        self.mcm_couchdb_url = mcm_couchdb_url or os.getenv("MCM_COUCHDB_URL", "")
+        self.mcm_couchdb_credential = mcm_couchdb_credential or os.getenv(
+            "COUCH_CRED", ""
+        )
+        self.mcm_couchdb_lucene_url = mcm_couchdb_lucene_url or os.getenv(
+            "MCM_LUCENE_URL", ""
+        )
```

Comment:
I want to avoid `None` values as much as possible. That's the reason I prefer to set them to empty strings (also a falsy element) than using `None` directly.